File: sample-domain/npm/src/aggregates/parent/parent.ts
import { Child, IChild } from "./child";
import { AggregateBase } from "../../abstractions/aggregate";
import { IEntity } from "../../abstractions/entity";

export interface IParent extends IEntity {
    description: string;
    children: IChild[];
}

export class Parent extends AggregateBase {
    description: string;
    children: Child[];
    constructor(description: string) {
        super();
        this.description = description || "";
        this.children = [];
    }
    hydrate(state: Partial<IParent>): Parent {
        if (!state) {
            state = {};
        }
        this.id = state.id || 0;
        this.description = state.description || "";
        this.children = (state.children || []).map(x => new Child(0, "").hydrate(x));
        return this;
    }
    addChild(description: string) {
        const child = new Child(this.id, description);
        this.children.push(child);
        return child;
    }
    removeChild(child: Child) {
        const found = this.children.indexOf(child);
        if (found >= 0) {
            this.children.splice(found, 1);
        }
    }
}

File: sample-domain/dotnet/src/aggregates/parent/parent.cs
using System.Collections.Generic;
using Sample.Domain.Abstractions;

namespace Sample.Domain.Aggregates.Parent
{
    public class Parent : AggregateBase, IAggregate
    {
        public virtual string Description { get; set; }
        public virtual IList<Child> Children { get; set; }
        private Parent() { }
        public Parent(string description) 
        {
            Description = description;
        }
        public Child AddChild(string description) 
        {
            var child = new Child(Id, description);
            Children.Add(child);
            return child;
        }
        public void RemoveChild(Child child)
        {
            if (Children.Contains(child))
                Children.Remove(child);
        }
    }
}